### PR TITLE
update the prompt to include `needs-info`

### PR DESCRIPTION
**Pull request by devoncarew**
- update the prompt to include triaging issues w/ `needs-info` labels
- `[breaking change]` issues should have `breaking-change-request` labels but don't necessarily need area labels
- add 5 'few-shot' examples in the classification prompt
- update benchmark expectations
- classification performance has generally improved from 56% to 65%

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/sdk_triage_bot/lib/src/gemini.dart [hunk 1]
```diff
@@ -43,7 +44,7 @@ class GeminiService {
   /// On failures, this will throw a `GenerativeAIException`.
   Future<List<String>> classify(String prompt) async {
     final result = await _query(_classifyModel, prompt);
-    final labels = result.split(',').map((l) => l.trim()).toList();
+    final labels = result.split(',').map((l) => l.trim()).toList()..sort();
```
Comment (mosuem) [inline pkgs/sdk_triage_bot/lib/src/gemini.dart:47]: I personally like [`sorted`](https://api.flutter.dev/flutter/package-collection_collection/IterableExtension/sorted.html) more - but it's the same under the hood.

File: pkgs/sdk_triage_bot/lib/src/gemini.dart [hunk 2]
```diff
@@ -7,6 +7,7 @@ import 'package:http/http.dart' as http;
 
 class GeminiService {
   // gemini-1.5-pro-latest, gemini-1.5-flash-latest, gemini-1.0-pro-latest
```
Comment (mosuem) [inline pkgs/sdk_triage_bot/lib/src/gemini.dart:9]: Maybe add a line on what these mean - probably `Possible values for models`?

File: pkgs/sdk_triage_bot/lib/src/gemini.dart [hunk 3]
```diff
@@ -43,7 +44,7 @@ class GeminiService {
   /// On failures, this will throw a `GenerativeAIException`.
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/lib/src/gemini.dart:44]: Maybe make `[GenerativeAIException]` a link.

File: pkgs/sdk_triage_bot/lib/src/prompts.dart [hunk 4]
```diff
@@ -45,13 +47,110 @@ If the issue is clearly a bug report, then also apply the label 'type-bug'.
 If the issue is mostly a question,  then also apply the label 'type-question'.
 Otherwise don't apply a 'type-' label.
 
+If the issue title starts with "[breaking change]" then apply the
+`breaking-change-request` label but don't assign an area label. IMPORTANT: only
+do this if the issue title starts with "[breaking change]".
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/lib/src/prompts.dart:52]: Tricky. I think we actually *shouldn't* add the tag.

Users can choose to write `[breaking change]`, but that doesn't make it a breaking change request. *Real* breaking change requests come from the template, which should set labels too. That means that if the `breaking-change-request` label isn't already set, this is probably *not* a real breaking change request, just a user cargo-culting our process, and what they have is an enhancement request, likely one that does need an area.

File: pkgs/sdk_triage_bot/lib/src/prompts.dart [hunk 5]
```diff
@@ -45,13 +47,110 @@ If the issue is clearly a bug report, then also apply the label 'type-bug'.
...
+If the issue was largely unchanged from our default issue template, then apply the
+'needs-info' label and don't assign an area label. These issues will generally
+have a title of "Create an issue" and the body will start with
+"Thank you for taking the time to file an issue!".
+
+If the issue title is "Analyzer Feedback from IntelliJ", these are generally not
+well qualified. For these issues, apply the 'needs-info' label but don't assign
+an area label.
+
 Return the labels as comma separated text.
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/lib/src/prompts.dart:63]: "Returns"

(A section starting with "Returns" is the recommended way to describe the returned value, just like a section starting with "The \[foo\] value ..." referencing a parameter is the recommended way to document that parameter.)

File: pkgs/sdk_triage_bot/lib/triage.dart [hunk 6]
```diff
@@ -141,7 +149,9 @@ ${trimmedBody(comment.body ?? '')}
...
 }
 
-List<String> filterExistingLabels(
-    List<String> allLabels, List<String> newLabels) {
-  return newLabels.toSet().intersection(allLabels.toSet()).toList();
+List<String> filterLegalLabels(
+  List<String> labels, {
+  required List<String> allRepoLabels,
+}) {
+  return labels.toSet().intersection(allRepoLabels.toSet()).toList()..sort();
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/lib/triage.dart:156]: Overkill to do `toSet()` on the source.
Consider:
```dart
final validLabels = allRepoLabels.toSet();
return [for (var label in labels) 
    if (validLabels.contains(label)) label]..sort();
```

File: pkgs/sdk_triage_bot/tool/bench.dart [hunk 7]
```diff
@@ -114,8 +109,22 @@ class ClassificationResults {
   }
 
   bool satisfiedBy(List<String> labels) {
-    final filtered = labels.where((l) => !l.startsWith('type-')).toSet();
-    final expected = expectedLabels.where((l) => !l.startsWith('type-'));
-    return expected.every(filtered.contains);
+    // handle needs-info
+    if (expectedLabels.contains('needs-info')) {
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/tool/bench.dart:113]: (I assume this is for testing/quality control only, so the repeated linear list lookups shouldn't be a problem. Right?)

File: pkgs/sdk_triage_bot/lib/src/common.dart [hunk 8]
```diff
@@ -37,9 +37,9 @@ String get geminiKey {
   return token;
 }
 
-/// Don't return more than 5k of text for an issue body.
+/// Don't return more than 10k of text for an issue body.
```
Comment (lrhn) [inline pkgs/sdk_triage_bot/lib/src/common.dart:40]: Say what it does, not what it doesn't.

Consider:
```dart
/// Maximal length of body used for querying.
const bodyLengthLimit = 10 * 1024;

/// The [body], truncated if larger than [bodyLengthLimit].
String trimmedBody(String body) { 
  // ...
```

**Review by lrhn — APPROVED** [review]

File: pkgs/sdk_triage_bot/lib/src/prompts.dart (hunk 4 above)
Comment (devoncarew) [reply pkgs/sdk_triage_bot/lib/src/prompts.dart]: I think I agree. I'll change this so the bot avoids triaging `[breaking change]` issues; manual review can figure out if they actually need any action.

**Pull request merged**